pkg/core/workflow_store.go

```diff
@@ -62,44 +62,85 @@ func (ws *WorkflowStore) AddWorkflow(ctx context.Context, w *workflow.Workflow)
-func (ws *WorkflowStore) GetAssignments(ctx context.Context, workflowName *string) (result map[string][]*domain.Codeset) {
-	result = make(map[string][]*domain.Codeset, len(ws.items))
+func (ws *WorkflowStore) GetAssignments(ctx context.Context, workflowName *string) (result map[string][]*domain.AssignedCodeset) {
+	result = make(map[string][]*domain.AssignedCodeset, len(ws.items))
 	if workflowName != nil {
-		if _, exists := ws.items[*workflowName]; exists {
-			result[*workflowName] = ws.items[*workflowName].assignedCodesets
+		if sw, exists := ws.items[*workflowName]; exists && len(sw.assignedCodesets) > 0 {
+			result[*workflowName] = sw.assignedCodesets
 		}
 		return
 	}
 	for _, sw := range ws.items {
-		result[sw.workflow.Name] = sw.assignedCodesets
+		if len(sw.assignedCodesets) > 0 {
+			result[sw.workflow.Name] = sw.assignedCodesets
+		}
 	}
 	return
 }
 
 // AddCodesetAssignment adds a codeset to the list of assigned codesets of a workflow if it does not already exists
-func (ws *WorkflowStore) AddCodesetAssignment(ctx context.Context, workflowName string, codeset *domain.Codeset) []*domain.Codeset {
+func (ws *WorkflowStore) AddCodesetAssignment(ctx context.Context, workflowName string,
+	assignedCodeset *domain.AssignedCodeset) []*domain.AssignedCodeset {
 	assignedCodesets := ws.items[workflowName].assignedCodesets
-	if !containsCodeset(assignedCodesets, codeset) {
-		assignedCodesets = append(assignedCodesets, codeset)
+	if assigned, _ := getAssignedCodeset(assignedCodesets, assignedCodeset.Codeset); assigned == nil {
+		assignedCodesets = append(assignedCodesets, assignedCodeset)
 		ws.items[workflowName].assignedCodesets = assignedCodesets
 	}
 	return assignedCodesets
 }
 
-func containsCodeset(slice []*domain.Codeset, codeset *domain.Codeset) bool {
-	for _, c := range slice {
-		if c.Project == codeset.Project && c.Name == codeset.Name {
-			return true
+// DeleteCodesetAssignment deletes a codeset from the list of assigned codesets of a workflow if it exists
+func (ws *WorkflowStore) DeleteCodesetAssignment(ctx context.Context, workflowName string, codeset *domain.Codeset) []*domain.AssignedCodeset {
+	assignedCodesets := ws.items[workflowName].assignedCodesets
+	if _, i := getAssignedCodeset(assignedCodesets, codeset); i != -1 {
+		assignedCodesets = removeAssignedCodeset(assignedCodesets, i)
+		ws.items[workflowName].assignedCodesets = assignedCodesets
+	}
+	return assignedCodesets
+}
+
+// GetAssignedCodeset returns a AssignedCodeset for the Workflow and Codeset
+func (ws *WorkflowStore) GetAssignedCodeset(ctx context.Context, workflowName string, codeset *domain.Codeset) *domain.AssignedCodeset {
+	sw, exists := ws.items[workflowName]
+	if !exists {
+		return nil
+	}
+	ac, _ := getAssignedCodeset(sw.assignedCodesets, codeset)
+	return ac
+}
+
+func getAssignedCodeset(slice []*domain.AssignedCodeset, codeset *domain.Codeset) (*domain.AssignedCodeset, int) {
+	for i, ac := range slice {
```

Comment:
is it wise to use variable with the name of the type?

Reply:
done